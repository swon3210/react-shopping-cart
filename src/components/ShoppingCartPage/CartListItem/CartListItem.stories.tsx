import CartListItem, { Props } from './CartListItem';

export default {
  title: 'Components/ShoppingCartPage/CartItem',
  component: CartListItem,
  argTypes: {},
};

const Template = (args: Props) => <CartListItem {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  name: '[든든] 야채바삭 김말이 700g',
  price: '18,900',
};
