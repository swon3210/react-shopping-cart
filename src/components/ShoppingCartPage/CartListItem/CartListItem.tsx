import * as Styled from './CartListItem.styles';
import Checkbox from '../../commons/Checkbox/Checkbox';
import trashCanSVG from '../../../assets/svgs/trash-can.svg';
import noImagePNG from '../../../assets/images/no-image.png';
import { CartItem } from '../../../type';
import ProductQuantityInput from '../ProductQuantityInput/ProductQuantityInput';

export interface Props {
  name: string;
  price: string;
  thumbnail?: string;
  quantity: CartItem['quantity'];
  setQuantity: (quantity: CartItem['quantity']) => void;
  isSelected: boolean;
  setSelected: (isSelected: CartItem['isSelected']) => void;
  onCartItemDelete: () => void;
}

const CartListItem = ({
  name,
  price,
  thumbnail = noImagePNG,
  quantity,
  setQuantity,
  isSelected,
  setSelected,
  onCartItemDelete,
}: Props) => {
  return (
    <Styled.CartListItem>
      <Checkbox checked={isSelected} onCheck={setSelected} />
      <Styled.Thumbnail src={thumbnail} alt="cart item thumbnail" />
      <Styled.ItemContentWrapper>
        <Styled.ItemContentTop>
          <Styled.ItemName>{name}</Styled.ItemName>
          <Styled.DeleteIcon src={trashCanSVG} alt="cart item delete button" onClick={onCartItemDelete} />
        </Styled.ItemContentTop>
        <Styled.ItemContentBottom>
          <ProductQuantityInput quantity={quantity} setQuantity={setQuantity} maxNumber={99} />
          <Styled.Price>{price}Ïõê</Styled.Price>
        </Styled.ItemContentBottom>
      </Styled.ItemContentWrapper>
    </Styled.CartListItem>
  );
};

export default CartListItem;
