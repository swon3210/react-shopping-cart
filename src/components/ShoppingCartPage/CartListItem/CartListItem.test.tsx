import CartListItem from './CartListItem';
import { render } from '@testing-library/react';

describe('CartListItem Component', () => {
  it('CartListItem Snapshot', () => {
    const onCartItemDelete = () => {
      console.log('onCartItemDelete');
    };

    const setQuantity = () => {
      console.log('setQuantity');
    };

    const setSelected = () => {
      console.log('setQuantity');
    };

    const cartListItemUtil = render(
      <CartListItem
        name="이름"
        price="2000"
        isSelected={true}
        onCartItemDelete={onCartItemDelete}
        quantity="2"
        setQuantity={setQuantity}
        setSelected={setSelected}
      ></CartListItem>
    );

    expect(cartListItemUtil.asFragment()).toMatchSnapshot();
  });
});
