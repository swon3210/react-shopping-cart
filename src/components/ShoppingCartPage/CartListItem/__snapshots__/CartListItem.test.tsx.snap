// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartListItem Component CartListItem Snapshot 1`] = `
<DocumentFragment>
  <li
    class="sc-bdvvaa hBafqH"
  >
    <div
      class="sc-furvIG haXEwx"
    >
      <input
        checked=""
        class="sc-pVTma ejcax"
        type="checkbox"
      />
    </div>
    <img
      alt="cart item thumbnail"
      class="sc-gsDJrp kDJhwx"
      src="no-image.png"
    />
    <div
      class="sc-dkPtyc eUFjw"
    >
      <div
        class="sc-hKwCoD isHNnm"
      >
        <span
          class="sc-jRQAMF jKDpyP"
        >
          이름
        </span>
        <img
          alt="cart item delete button"
          class="sc-gKckTs ldQxvY"
          src="trash-can.svg"
        />
      </div>
      <div
        class="sc-eCImvq cJiCUx"
      >
        <div
          class="sc-kDThTU guPzNQ"
        >
          <input
            class="sc-iqsfdx dmOsdq"
            type="number"
            value="2"
          />
          <div
            class="sc-crHlIS gThUof"
          >
            <button
              class="sc-egiSv kWYJHJ"
            >
              <img
                alt="수량 증가 버튼"
                src="triangle-up.svg"
              />
            </button>
            <button
              class="sc-egiSv kWYJHJ"
            >
              <img
                alt="수량 감소 버튼"
                src="triangle-down.svg"
              />
            </button>
          </div>
        </div>
        <span
          class="sc-iCfLBT iEZWhA"
        >
          2000원
        </span>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
